// @fluentui/react-tree/lib-commonjs/components/TreeItemPersonaLayout/useTreeItemPersonaLayout.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useTreeItemPersonaLayout_unstable", {
    enumerable: true,
    get: ()=>useTreeItemPersonaLayout_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _contexts = require("../../contexts");
const _tokens = require("../../utils/tokens");
const _useTreeItemLayout = require("../TreeItemLayout/useTreeItemLayout");
const useTreeItemPersonaLayout_unstable = (props, ref)=>{
    const { media , content , children , main , description  } = props;
    const treeItemLayoutState = (0, _useTreeItemLayout.useTreeItemLayout_unstable)({
        ...props,
        iconBefore: null,
        iconAfter: null
    }, ref);
    const size = (0, _contexts.useTreeContext_unstable)((ctx)=>ctx.size);
    return {
        ...treeItemLayoutState,
        components: {
            expandIcon: 'div',
            content: 'div',
            main: 'div',
            description: 'div',
            root: 'div',
            media: 'div'
        },
        avatarSize: _tokens.treeAvatarSize[size],
        main: (0, _reactUtilities.resolveShorthand)(main, {
            required: true,
            defaultProps: {
                children
            }
        }),
        media: (0, _reactUtilities.resolveShorthand)(media, {
            required: true
        }),
        content: (0, _reactUtilities.resolveShorthand)(content, {
            required: true
        }),
        description: (0, _reactUtilities.resolveShorthand)(description)
    };
};
