// @fluentui/react-tree/lib-commonjs/components/TreeItemPersonaLayout/useTreeItemPersonaLayoutStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    treeItemPersonaLayoutClassNames: ()=>treeItemPersonaLayoutClassNames,
    useTreeItemPersonaLayoutStyles_unstable: ()=>useTreeItemPersonaLayoutStyles_unstable
});
const _react = require("@griffel/react");
const _treeItemContext = require("../../contexts/treeItemContext");
const treeItemPersonaLayoutClassNames = {
    root: 'fui-TreeItemPersonaLayout',
    expandIcon: 'fui-TreeItemPersonaLayout__expandIcon',
    media: 'fui-TreeItemPersonaLayout__media',
    content: 'fui-TreeItemPersonaLayout__content',
    description: 'fui-TreeItemPersonaLayout__description',
    main: 'fui-TreeItemPersonaLayout__main'
};
/**
 * Styles for the root slot
 */ const useRootStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        Bahqtrf: "fk6fouc",
        Be2twd7: "fkhj508",
        Bhrd7zp: "figsok6",
        Bg96gwp: "f1i3iumi",
        Ijaq50: "f15ws6j",
        Br312pm: "f135tdbu",
        nk6f5a: "f2e2169",
        Bw0ie65: "f4rqp6x",
        lj723h: "flvvhsy",
        ecr2s2: "f1wfn5kd",
        zt0xaq: "f5na5aj",
        Bi91k9c: "fnwyq0v",
        Jwef8y: "f1t94bn6",
        tbva4x: "f1oboesa"
    },
    leaf: {
        uwmqm3: [
            "f1k1erfc",
            "faevyjx"
        ]
    },
    branch: {
        uwmqm3: [
            "fo100m9",
            "f6yw3pu"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".f1i3iumi{line-height:var(--lineHeightBase300);}",
        ".f15ws6j{grid-row-start:layout;}",
        ".f135tdbu{grid-column-start:layout;}",
        ".f2e2169{grid-row-end:layout;}",
        ".f4rqp6x{grid-column-end:layout;}",
        ".f1k1erfc{padding-left:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}",
        ".faevyjx{padding-right:calc(var(--fluent-TreeItem--level, 1) * var(--spacingHorizontalXXL));}",
        ".fo100m9{padding-left:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}",
        ".f6yw3pu{padding-right:calc((var(--fluent-TreeItem--level, 1) - 1) * var(--spacingHorizontalXXL));}"
    ],
    a: [
        ".flvvhsy:active{color:var(--colorNeutralForeground2Pressed);}",
        ".f1wfn5kd:active{background-color:var(--colorSubtleBackgroundPressed);}",
        ".f5na5aj:active .fui-TreeItemPersonaLayout__expandIcon{color:var(--colorNeutralForeground3Pressed);}"
    ],
    h: [
        ".fnwyq0v:hover{color:var(--colorNeutralForeground2Hover);}",
        ".f1t94bn6:hover{background-color:var(--colorSubtleBackgroundHover);}",
        ".f1oboesa:hover .fui-TreeItemPersonaLayout__expandIcon{color:var(--colorNeutralForeground3Hover);}"
    ]
});
/**
 * Styles for the expand icon slot
 */ const useMediaStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        a9b677: "f1szoe96",
        Bqenvij: "f1d2rq10",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "f7x41pl",
            "fruq291"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "fgiv446",
            "ffczdla"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f1szoe96{width:32px;}",
        ".f1d2rq10{height:32px;}",
        ".f1g0x7ka{padding-top:0;}",
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".f1qch9an{padding-bottom:0;}",
        ".fgiv446{padding-left:var(--spacingHorizontalXXS);}",
        ".ffczdla{padding-right:var(--spacingHorizontalXXS);}"
    ]
});
const useContentStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        a9b677: "fly5x3f",
        mc9l5x: "f22iagw",
        Beiy3e4: "f1vx9l62",
        z8tnut: "f1ngh7ph",
        z189sj: [
            "f7x41pl",
            "fruq291"
        ],
        Byoj8tv: "f5o476b",
        uwmqm3: [
            "f1f5gg8d",
            "f1vdfbxk"
        ],
        i8kkvl: "f1fps8zv",
        Belr9w4: "fws515f"
    }
}, {
    d: [
        ".fly5x3f{width:100%;}",
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f1vx9l62{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}",
        ".f1ngh7ph{padding-top:var(--spacingVerticalMNudge);}",
        ".f7x41pl{padding-right:var(--spacingHorizontalXS);}",
        ".fruq291{padding-left:var(--spacingHorizontalXS);}",
        ".f5o476b{padding-bottom:var(--spacingVerticalMNudge);}",
        ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}",
        ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}",
        ".f1fps8zv{-webkit-column-gap:var(--spacingVerticalNone);column-gap:var(--spacingVerticalNone);}",
        ".fws515f{row-gap:var(--spacingHorizontalNone);}"
    ]
});
const useDescriptionStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        Bahqtrf: "fk6fouc",
        Be2twd7: "fy9rknc",
        Bhrd7zp: "figsok6",
        Bg96gwp: "fwrc4pm"
    }
}, {
    d: [
        ".fk6fouc{font-family:var(--fontFamilyBase);}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".figsok6{font-weight:var(--fontWeightRegular);}",
        ".fwrc4pm{line-height:var(--lineHeightBase200);}"
    ]
});
/**
 * Styles for the expand icon slot
 */ const useExpandIconStyles = /*#__PURE__*/ (0, _react["__styles"])({
    base: {
        mc9l5x: "f22iagw",
        Bt984gj: "f122n59",
        Brf1p80: "f4d9j23",
        Bf4jedk: "f17fgpbq",
        B7ck84d: "f1ewtqcl",
        sj55zd: "f11d4kpn",
        Bh6795r: "f1jhi6b8",
        Bnnss6s: "fi64zpg",
        xawz: "f1rmlqtg",
        z8tnut: "f1ywm7hm",
        z189sj: [
            "fhxju0i",
            "f1cnd47f"
        ],
        Byoj8tv: "f14wxoun",
        uwmqm3: [
            "f1cnd47f",
            "fhxju0i"
        ]
    }
}, {
    d: [
        ".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
        ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
        ".f4d9j23{-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;}",
        ".f17fgpbq{min-width:24px;}",
        ".f1ewtqcl{box-sizing:border-box;}",
        ".f11d4kpn{color:var(--colorNeutralForeground3);}",
        ".f1jhi6b8{-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;}",
        ".fi64zpg{-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;}",
        ".f1rmlqtg{-webkit-flex-basis:auto;-ms-flex-preferred-size:auto;flex-basis:auto;}",
        ".f1ywm7hm{padding-top:var(--spacingVerticalXS);}",
        ".fhxju0i{padding-right:0;}",
        ".f1cnd47f{padding-left:0;}",
        ".f14wxoun{padding-bottom:var(--spacingVerticalXS);}"
    ]
});
const useTreeItemPersonaLayoutStyles_unstable = (state)=>{
    const rootStyles = useRootStyles();
    const mediaStyles = useMediaStyles();
    const contentStyles = useContentStyles();
    const descriptionStyles = useDescriptionStyles();
    const expandIconStyles = useExpandIconStyles();
    const itemType = (0, _treeItemContext.useTreeItemContext_unstable)((ctx)=>ctx.itemType);
    state.root.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.root, rootStyles.base, rootStyles[itemType], state.root.className);
    state.media.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.media, mediaStyles.base, state.media.className);
    if (state.content) {
        state.content.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.content, contentStyles.base, state.content.className);
    }
    if (state.main) {
        state.main.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.main, state.main.className);
    }
    if (state.description) {
        state.description.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.description, descriptionStyles.base, state.description.className);
    }
    if (state.expandIcon) {
        state.expandIcon.className = (0, _react.mergeClasses)(treeItemPersonaLayoutClassNames.expandIcon, expandIconStyles.base, state.expandIcon.className);
    }
    return state;
}; //# sourceMappingURL=useTreeItemPersonaLayoutStyles.styles.js.map
