// @fluentui/react-drawer/lib-commonjs/components/DrawerOverlay/useDrawerOverlayStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    drawerOverlayClassNames: ()=>drawerOverlayClassNames,
    useDrawerOverlayStyles_unstable: ()=>useDrawerOverlayStyles_unstable
});
const _react = require("@griffel/react");
const _useDrawerBaseStylesStyles = require("../../util/useDrawerBaseStyles.styles");
const drawerOverlayClassNames = {
    root: 'fui-DrawerOverlay'
};
/**
 * Styles for the root slot
 */ const useStyles = /*#__PURE__*/ (0, _react["__styles"])({
    root: {
        qhf8xq: "f19dog8a",
        Bhzewxz: "f15twtuk",
        B5kzvoi: "f1yab3r1"
    }
}, {
    d: [
        ".f19dog8a{position:fixed;}",
        ".f15twtuk{top:0;}",
        ".f1yab3r1{bottom:0;}"
    ]
});
const useDrawerOverlayStyles_unstable = (state)=>{
    const baseStyles = (0, _useDrawerBaseStylesStyles.useDrawerBaseStyles)();
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(drawerOverlayClassNames.root, baseStyles.root, styles.root, state.size && baseStyles[state.size], state.position && baseStyles[state.position], state.root.className);
    return state;
}; //# sourceMappingURL=useDrawerOverlayStyles.styles.js.map
