// @fluentui/react-tree/lib-commonjs/components/TreeItemAside/useTreeItemAside.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useTreeItemAside_unstable", {
    enumerable: true,
    get: ()=>useTreeItemAside_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _treeItemContext = require("../../contexts/treeItemContext");
const useTreeItemAside_unstable = (props, ref)=>{
    const actionsRef = (0, _treeItemContext.useTreeItemContext_unstable)((ctx)=>ctx.actionsRef);
    const contextVisible = (0, _treeItemContext.useTreeItemContext_unstable)((ctx)=>props.actions ? ctx.isActionsVisible : ctx.isAsideVisible);
    const { actions =false , visible =contextVisible  } = props;
    return {
        actions,
        visible,
        buttonContextValue: {
            size: 'small'
        },
        components: {
            root: 'div'
        },
        root: (0, _reactUtilities.getNativeElementProps)('div', {
            ref: (0, _reactUtilities.useMergedRefs)(ref, actions ? actionsRef : undefined),
            ...props
        })
    };
};
