// @fluentui/react-virtualizer/lib-commonjs/components/VirtualizerScrollViewDynamic/useVirtualizerScrollViewDynamic.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "useVirtualizerScrollViewDynamic_unstable", {
    enumerable: true,
    get: ()=>useVirtualizerScrollViewDynamic_unstable
});
const _interopRequireWildcard = require("@swc/helpers/lib/_interop_require_wildcard.js").default;
const _react = /*#__PURE__*/ _interopRequireWildcard(require("react"));
const _reactUtilities = require("@fluentui/react-utilities");
const _useVirtualizer = require("../Virtualizer/useVirtualizer");
const _hooks = require("../../Hooks");
const _utilities = require("../../Utilities");
function useVirtualizerScrollViewDynamic_unstable(props) {
    const contextState = (0, _utilities.useVirtualizerContextState_unstable)(props.virtualizerContext);
    var _props_axis, _contextState_contextIndex;
    const { virtualizerLength , bufferItems , bufferSize , scrollRef  } = (0, _hooks.useDynamicVirtualizerMeasure)({
        defaultItemSize: props.itemSize,
        direction: (_props_axis = props.axis) !== null && _props_axis !== void 0 ? _props_axis : 'vertical',
        getItemSize: props.getItemSize,
        currentIndex: (_contextState_contextIndex = contextState === null || contextState === void 0 ? void 0 : contextState.contextIndex) !== null && _contextState_contextIndex !== void 0 ? _contextState_contextIndex : 0,
        numItems: props.numItems
    });
    const iScrollRef = (0, _reactUtilities.useMergedRefs)(_react.useRef(null), scrollRef);
    const virtualizerState = (0, _useVirtualizer.useVirtualizer_unstable)({
        ...props,
        virtualizerLength,
        bufferItems,
        bufferSize,
        scrollViewRef: iScrollRef,
        virtualizerContext: contextState
    });
    return {
        ...virtualizerState,
        components: {
            ...virtualizerState.components,
            container: 'div'
        },
        container: (0, _reactUtilities.resolveShorthand)(props.container, {
            required: true,
            defaultProps: {
                ref: iScrollRef
            }
        })
    };
}
