// @fluentui/react-tree/lib/components/TreeItemAside/useTreeItemAsideStyles.styles.js
import { __styles, mergeClasses, shorthands } from '@griffel/react';
import { tokens } from '@fluentui/react-theme';
export const treeItemAsideClassNames = {
  root: 'fui-TreeItemAside'
};
/**
 * Styles for the action icon slot
 */
const useStyles = /*#__PURE__*/__styles({
  base: {
    mc9l5x: "f22iagw",
    Frg6f3: ["fcgxt0o", "f1ujusj6"],
    Ijaq50: "fobksn0",
    Br312pm: "fmy5l6f",
    nk6f5a: "fzqypwc",
    Bw0ie65: "f1tmftl3"
  },
  actions: {
    qhf8xq: "f10pi13n",
    Bj3rh1h: "f19g0ac",
    z8tnut: "f1g0x7ka",
    z189sj: ["f1vdfbxk", "f1f5gg8d"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["f1f5gg8d", "f1vdfbxk"]
  },
  aside: {
    Bt984gj: "f122n59",
    Bj3rh1h: "f11zp4z2",
    z8tnut: "f1g0x7ka",
    z189sj: ["fw5db7e", "f1uw59to"],
    Byoj8tv: "f1qch9an",
    uwmqm3: ["f1uw59to", "fw5db7e"],
    i8kkvl: "f1ufnopg",
    Belr9w4: "f14sijuj"
  }
}, {
  d: [".f22iagw{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}", ".fcgxt0o{margin-left:auto;}", ".f1ujusj6{margin-right:auto;}", ".fobksn0{grid-row-start:aside;}", ".fmy5l6f{grid-column-start:aside;}", ".fzqypwc{grid-row-end:aside;}", ".f1tmftl3{grid-column-end:aside;}", ".f10pi13n{position:relative;}", ".f19g0ac{z-index:1;}", ".f1g0x7ka{padding-top:0;}", ".f1vdfbxk{padding-right:var(--spacingHorizontalS);}", ".f1f5gg8d{padding-left:var(--spacingHorizontalS);}", ".f1qch9an{padding-bottom:0;}", ".f122n59{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}", ".f11zp4z2{z-index:0;}", ".fw5db7e{padding-right:var(--spacingHorizontalM);}", ".f1uw59to{padding-left:var(--spacingHorizontalM);}", ".f1ufnopg{-webkit-column-gap:var(--spacingHorizontalXS);column-gap:var(--spacingHorizontalXS);}", ".f14sijuj{row-gap:var(--spacingHorizontalXS);}"]
});
/**
 * Apply styling to the TreeItemAside slots based on the state
 */
export const useTreeItemAsideStyles_unstable = state => {
  const styles = useStyles();
  state.root.className = mergeClasses(treeItemAsideClassNames.root, styles.base, state.actions ? styles.actions : styles.aside, state.root.className);
  return state;
};
